src/components/form-field/search-field/search-field.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useMemo, useRef } from "react";
+import { classNames } from "../../../util/class-names";
+import CrossIcon from "../../../icons/cross-icon";
+import SearchIcon from "../../../icons/search-icon";
+
+export interface SearchInputProps extends React.ComponentPropsWithoutRef<"input"> {
+    autoSelect?: boolean;
+    ariaDescribedBy?: string;
+    error?: boolean;
+    onClear: () => void;
+}
+
+export const SearchInput = ({
+    ariaDescribedBy,
+    readOnly,
+    autoSelect,
+    onClear,
+    error,
+    value,
+    disabled,
+    className,
+    ...props
+}: SearchInputProps) => {
+    const inputRef = useRef<HTMLInputElement>(null);
+
+    const isClearIconShown = useMemo(() => {
```

Comment:
The useMemo hook is useful for memoizing the result of a computation that is expensive and should not be re-computed on every render unless its dependencies have changed. In the case of isClearIconShown, the computation is simply a boolean check to determine if the clear icon should be shown. This check is neither computationally expensive nor does it result in a complex value that would benefit from memoization.
My suggestion would be:
`const isClearIconShown = !readOnly && !disabled && !!value;`